src/app/component/places/places.component.ts

```diff
@@ -0,0 +1,254 @@
+      }
+    ];
+
+    this.markerList = [
+      {
+        lat: 49.84579876141749,
+        lng: 24.025125288060412,
+        iconUrl: this.redIcon,
+        card: this.cardsCollection[0]
+      },
+      {
+        lat: 49.84423766639821,
+        lng: 24.02405838813841,
+        iconUrl: this.redIcon,
+        card: this.cardsCollection[1]
+      },
+      {
+        lat: 49.842959399097296,
+        lng: 24.026151152038356,
+        iconUrl: this.redIcon,
+        card: this.cardsCollection[2]
+      },
+      {
+        lat: 49.841649695791766,
+        lng: 24.026889537979542,
+        iconUrl: this.redIcon,
+        card: this.cardsCollection[3]
+      },
+      {
+        lat: 49.83929345961723,
+        lng: 24.025845314617513,
+        iconUrl: this.redIcon,
+        card: this.cardsCollection[4]
+      }
+    ];
+
+    this.markerListCopy = this.markerList.slice();
+
+    this.bindLang(this.localStorageService.getCurrentLanguage());
+  }
+
+  ngDoCheck(): void {
+    this.startSearch();
+  }
+
+  public getFilterData(tags: Array<string>): void {
+    if (tags.filter((item) => item === 'Saved places') && tags.length > 0) {
+      this.markerListCopy = this.markerList.filter((item) => {
+        return this.favoritePlaces.includes(item.card.id);
+      });
+    } else if (this.markerListCopy.length === 0) {
+      this.markerListCopy = this.markerList.slice();
+    }
+
+    this.tagList = ['Shops', 'Restaurants', 'Recycling points', 'Events', 'Saved places'];
+  }
+
+  private bindLang(lang: string): void {
+    this.translate.setDefaultLang(lang);
+  }
+
+  public getMarker(marker: AgmMarker): void {
+    this.markerList.forEach((item) => {
+      item.iconUrl = this.redIcon;
+    });
+    const clickedMarker = this.markerList.indexOf(
+      this.markerList.filter((item) => {
+        return item.lat === marker.latitude;
+      })[0]
+    );
+
+    this.markerList[clickedMarker].iconUrl = this.greenIcon;
+    this.contentObj = this.markerList[clickedMarker].card;
+    this.contentObj.cardStars = this.getStars(this.contentObj.cardRating);
+    // if(this.drawer.opened === true) {
+    //   this.drawer.toggle()
+    // }
+    this.drawer.toggle();
+  }
+
+  public moveToFavorite(event): void {
+    if (!event.toElement.src.includes(this.bookmarkSaved)) {
+      event.toElement.src = this.bookmarkSaved;
+      this.favoritePlaces.push(+event.toElement.parentNode.parentNode.id);
+    } else {
+      const indexToDelete = this.favoritePlaces.indexOf(event.toElement.parentNode.parentNode.id);
+      event.toElement.src = this.bookmark;
+      this.favoritePlaces.splice(indexToDelete, 1);
+    }
+  }
+
+  public startSearch(): void {
+    const searchParams = this.searchName.toLowerCase();
+    this.markerListCopy = this.markerList.filter((item) => {
+      return item.card.cardName.toLowerCase().includes(searchParams);
+    });
+  }
+
+  private getStars(rating: number): Array<string> {
+    if (rating > 5) {
```

Comment:
5 IMHO is magic number. And also 0.5(row#208)

Reply:
Roger that. Made them 'consts'.